src/network/index.js

```diff
@@ -1,14 +1,38 @@
 'use strict'
 
 module.exports = class Network {
-  constructor (libp2p) {
-    // TODO: Implement me
+  constructor (libp2p, peerBook, bitswap) {
     this.libp2p = libp2p
+    this.peerBook = peerBook
+
+    this.libp2p.swarm.handle('/ipfs/bitswap/1.0.0', (conn) => {
+      // TODO _receiveMessage(peerId, incoming)
+      // find out what is the peerId from the peer that opened this conn
+      // NEED: https://github.com/ipfs/go-ipfs/issues/2605
+
+      // TODO: _receiveError(err)
+    })
+
+    this.libp2p.swarm.on('peer-mux-established', (peerInfo) => {
+      bitswap._onPeerConnected(peerInfo.id)
+    })
+
+    this.libp2p.swarm.on('peer-mux-closed', (peerInfo) => {
+      bitswap._onPeerDisconnected(peerInfo.id)
+    })
   }
 
   // Connect to the given peer
   connectTo (peerId, cb) {
-    // TODO: Implement me
+    // NOTE: For now, all this does is ensure that we are
+    // connected. Once we have Peer Routing, we will be able
+    // to find the Peer
+    if (this.libp2p.swarm.muxedConns[peerId.toB58String()]) {
+      // TODO: ask - want _onPeerConnected to be called here too?
```

Comment:
@dignifiedquire should I call _onPeerConnected here too? Or is the `cb` enough to signal we successfully  dialed (I'm afraid that we fire to routines for connecting to the same peer)